### ISSUE TITLE
Clarify licensing of Singularity image files

### ISSUE THREAD

**Issue by remram44** [apptainer/singularity#1614]
I just pulled a Docker image with Singularity, opened it up and was surprised to see multiple copyright statement in the files under `.singularity.d/env`. Those are very simple "scripts" (technically; actually more lists of environment variables) but they contain copyright headers restricting the sharing of them, "including in binary form", without the accompanying license file.

You might already be breaking those license terms by creating image files without the full license text.

I am worried I am too during normal use of your software :disappointed:

Example text from `99-runtimevars.sh`:
```
# Copyright (c) 2017-2018, SyLabs, Inc. All rights reserved.
# 
# This software is licensed under a customized 3-clause BSD license.  Please
# consult LICENSE file distributed with the sources of this project regarding
# your rights to use or distribute this software.
```
This file explicitly mentions using a **customized** license which I simply do not know where to find.

`01-base.sh` is empty yet claims to be copyrighted by SyLabs, SingularityWare, Gregory Kurtzer, and the University of California, again under an unknown license. Some other files at least link to the license file in this GitHub repository.

This should be clarified, or more practically, those very simple scripts be put in the public domain. List of variables might not be copyrightable, but I am not a lawyer.

**Comment by jmstover** [comment]
@remram44 ,

@gmkurtzer will need to be the final authority here. But, the *original* License was basically a 3-Clause BSD _with_ a LBNL/DoE exemption. From the git repo, if you look at tag 1.0 and the LICENSE file within, you'll see the original license text (which I think is now LICENSE-LBNL.md).

But quite a bit of the code still falls under the original License.

**Comment by remram44** [comment]
Did you really mean for that license to cover the generated packages as well, or the program code?

I'm no stranger to license headers, and I can understand how those got here from you simply copying source files into images. But you can keep those in the software without letting them permeate the artifacts you generate with it.

gzip doesn't add a GNU license into gz files and I don't think you should either, those have real implications.

**Comment by jmstover** [comment]
Those files aren't all generated, some are copied directly... or stubs are used, and the stubs have it in them.

https://github.com/singularityware/singularity/blob/master/libexec/bootstrap-scripts/environment/99-runtimevars.sh

That's the 99-runtimevars.sh that's in the master branch (2.5.1). I believe (IIRC) that every provided file needed to have the license clause in it when under LBNL/DoE.

**Comment by remram44** [comment]
So not only the packages, but the Singularity source distribution is illegally including this file without the accompanying LICENSE file (or is it one of LICENSE.md or LICENSE-LBNL.md in this repo?)

**Comment by jmstover** [comment]
The text everywhere wasn't updated unfortunately. One of the specifics that got over looked.

The LICENSE file being referenced in this case would be the LICENSE-LBNL.md. That's the customized BSD license with the explicit references to LBNL and Department of Energy, and the original license that Singularity was released under.

It was modified to the straight BSD license when it was no longer being partially funded by the DoE. But, some code hasn't been touched since the move, and changing the license text isn't one of the things we think about consciously. The custom one (LICENSE-LBNL.md) has an **explicit** mention of LBNL and DoE, and that is really the biggest difference. It specifically mentions those organizations, where the normal BSD 3-clause doesn't.

**Comment by gmkurtzer** [comment]
This is a good point, thank you for bringing it up @remram44! The files have the default header included in all of our code, but I don't believe they have any of the legacy code created under the modified BSD (LBNL-license) within them. We can check that easily enough and remove that bit if it no longer applies.

**Comment by remram44** [comment]
Bump! Work on this seems to have stopped. If you tell me about your decision, maybe I can try and implement it?

**Comment by jmstover** [comment]
Hi @remram44 .  With the 3.0 rewrite, we're completely under the BSD 3-clause license. There should no longer be any code under the modified LBNL license.

@gmkurtzer ... if that's the case then can we remove the LICENSE-LBNL.md file from the repository for the 3.0 branch? Or does it need to stay for legacy reasons?

**Comment by carterpeel** [comment]
Hello,

This is a templated response that is being sent out to all open issues.  We are working hard on 'rebuilding' the Singularity community, and a major task on the agenda is finding out what issues are still outstanding.


**Please consider the following:**

1. Is this issue a duplicate, or has it been fixed/implemented since being added?
2. Is the issue still relevant to the current state of Singularity's functionality?
3. Would you like to continue discussing this issue or feature request?


Thanks,
Carter

**Comment by remram44** [comment]
This issue has been manually marked as not stale by me.

**Comment by pedroalvesbatista** [comment]
@remram44 Our team will discuss this in a bit, and return with a more polished solution for this. Thankyou for keeping up with the interest in this issue.

**Comment by kmuriki** [comment]
Closing this old issue in the old Singularity repo. If its still relevant please create a new issue in the Apptainer repo. Thanks.

**Comment by remram44** [comment]
This should probably still be clarified because it affects the licensing of IMAGE FILES created with the old version. Even if you replace the software, most certainly those images are still around.

Can you please confirm that those license claims are ERRONEOUS and can be legally ignored with regard to the handling of those images files, thanks.

**Comment by DrDaveD** [comment]
Clearly there's nothing that can be done about previously generated images.

It seems to me the simplest thing is to put in references in each of these files to the license files in the github source repository.  Does anybody object to that?

**Comment by remram44** [comment]
There's nothing that can be done about *the content* of those images, but you can still clear up the licensing issue. For example here. You could say "yes this was a mistake, the image are not tainted by this license even though we can't change the text inside", which would allow me and everyone finding this issue to move forward.

It's been 4 years. If you can't figure out the legal situation of your own code in that time, how can you expect users to depend on it and trust you?

**Comment by DrDaveD** [comment]
I don't think it's a matter of "can't" figure out the legal situation; this is the first I have seen the issue but I suspect that it's more a matter that nobody has considered it a high enough priority to do something about it.

**Comment by remram44** [comment]
Again, I am not asking anyone to do something about it. I am asking for confirmation that images created by Singularity could ever be shared legally. All you have to say is "yes", and the fact that 4 years and 15 messages later no one has said it yet is becoming really worrying.

I really thought this would be a formality but you are giving all the signs that something legally dodgy is going on.

**Comment by DrDaveD** [comment]
Probably no-one has wanted to say it because they're not lawyers and they didn't think it was worth the effort to contact one.

I think I can say with confidence that none of the copyright holders intend to restrict anyone from sharing images.  They just want the code they've written to retain their copyrights if it gets reused in some other way.  At this point with all the different copyright holders it's probably not an option to put those files in the public domain, so I think the fix that I propose in apptainer/apptainer#497 should be an easy and reasonable compromise.
